helix-lock/src/main/java/org/apache/helix/lock/HelixLock.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.helix.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Generic interface for Helix distributed lock
+ */
+public interface HelixLock {
+  /**
+   * Acquire a lock
+   * @return true if the lock was acquired, false if could not be acquired
+   */
+  public boolean acquireLock();
+
+  /**
+   * Release a lock
+   * @return true if the lock was released, false if it could not be released
+   */
+  public boolean releaseLock();
+
+  /**
+   * Retrieve the lock information, e.g. lock timeout, lock message, etc.
+   * @return lock information
+   */
+  public Object getLockInfo();
```

Comment:
should this getLockInfo return a defined type (for example, LockInfo)?    Also, how does a piece of code check whether it owns the lock or not?

Reply:
Good point. I'll change the return type to LockInfo, which could these fields for now: owner ID, expiring time, lock message.
Since the lock info already have the owner ID, user can check if they are the lock owner or not by comparing their ID to the ID returned in LockInfo. If we want to make it easier for users, we can take in the user ID in the lock constructor and when they call isOwner() API, we compare the user ID with  the owner ID, and return true or false.

Reply:
Please try to make LockInfo very generic, as we will have various implementations, and we cannot force all of them to have certain fields.

Reply:
Added LockInfo interface as the type contains lock metadata. And I also added a isOwner method for customers to check if they are the owner of the lock.